homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump.py

```diff
@@ -0,0 +1,300 @@
+"""Support for HitachiAirToAirHeatPump."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState, Protocol
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    PRESET_NONE,
+    SWING_BOTH,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    SWING_VERTICAL,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+PRESET_HOLIDAY_MODE = "holiday_mode"
+
+FAN_SPEED_STATE = {
+    Protocol.OVP: OverkizState.OVP_FAN_SPEED,
+    Protocol.HLRR_WIFI: OverkizState.HLRRWIFI_FAN_SPEED,
+}
+
+LEAVE_HOME_STATE = {
+    Protocol.OVP: OverkizState.OVP_LEAVE_HOME,
+    Protocol.HLRR_WIFI: OverkizState.HLRRWIFI_LEAVE_HOME,
+}
+
+MAIN_OPERATION_STATE = {
+    Protocol.OVP: OverkizState.OVP_MAIN_OPERATION,
+    Protocol.HLRR_WIFI: OverkizState.HLRRWIFI_MAIN_OPERATION,
+}
+
+MODE_CHANGE_STATE = {
+    Protocol.OVP: OverkizState.OVP_MODE_CHANGE,
+    Protocol.HLRR_WIFI: OverkizState.HLRRWIFI_MODE_CHANGE,
+}
+
+ROOM_TEMPERATURE_STATE = {
+    Protocol.OVP: OverkizState.OVP_ROOM_TEMPERATURE,
+    Protocol.HLRR_WIFI: OverkizState.HLRRWIFI_ROOM_TEMPERATURE,
+}
+
+SWING_STATE = {
+    Protocol.OVP: OverkizState.OVP_SWING,
+    Protocol.HLRR_WIFI: OverkizState.HLRRWIFI_SWING,
+}
+
+OVERKIZ_TO_HVAC_MODES: dict[str, str] = {
+    OverkizCommandParam.AUTOHEATING: HVACMode.AUTO,
+    OverkizCommandParam.AUTOCOOLING: HVACMode.AUTO,
+    OverkizCommandParam.OFF: HVACMode.OFF,
+    OverkizCommandParam.HEATING: HVACMode.HEAT,
+    OverkizCommandParam.FAN: HVACMode.FAN_ONLY,
+    OverkizCommandParam.DEHUMIDIFY: HVACMode.DRY,
+    OverkizCommandParam.COOLING: HVACMode.COOL,
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+}
+
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_SWING_MODES: dict[str, str] = {
+    OverkizCommandParam.BOTH: SWING_BOTH,
+    OverkizCommandParam.HORIZONTAL: SWING_HORIZONTAL,
+    OverkizCommandParam.STOP: SWING_OFF,
+    OverkizCommandParam.VERTICAL: SWING_VERTICAL,
+}
+
+SWING_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_SWING_MODES.items()}
+
+# Fan modes differ between OVP and HLRRWIFI protocol
+OVERKIZ_TO_FAN_MODES: dict[Protocol, dict[str, str]] = {
+    Protocol.OVP: {
+        OverkizCommandParam.AUTO: FAN_AUTO,
+        OverkizCommandParam.HI: FAN_HIGH,
+        OverkizCommandParam.LO: FAN_LOW,
+        OverkizCommandParam.MED: FAN_MEDIUM,
+        OverkizCommandParam.SILENT: "silent",
```

Comment:
Maube you could use a constant for "silent" and use it here and in `FAN_MODES_TO_OVERKIZ`?

Reply:
I just fixed that :)